### ISSUE TITLE
Enable ECS ContainerDependency - DependsOn parameter

### ISSUE THREAD

**Issue by AlexJov** [ansible-collections/community.aws#520]
##### SUMMARY
Enabling init containers when describing Containers with ECS TaskDefinitions. 

##### ISSUE TYPE
- Feature Idea

##### COMPONENT NAME

ecs_taskdefinition

##### ADDITIONAL INFORMATION

Having container dependencies is supported by boto3 lib:
https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/ecs.html#ECS.Client.register_task_definition

However, at the moment is not part of the available parameters to pass through ansible:
https://github.com/ansible-collections/community.aws/blob/130cf3cc5980014020632f19fdab79c9bcf28add/plugins/modules/ecs_taskdefinition.py#L275

```
response = self.ecs.register_task_definition(**params)
```

What this feature would enable is having an init container to deliver binaries, for example monitoring agent (APM) files, before starting the essential container with the application monitored. Without `dependsOn` option available, in this case, the application starts before files are mounted from the agent's container and it fails to attach an agent.

**Comment by ansibullbot** [comment]
Files identified in the description:
* [`plugins/modules/ecs_taskdefinition.py`](https://github.com/['ansible-collections/amazon.aws', 'ansible-collections/community.aws', 'ansible-collections/community.vmware']/blob/main/plugins/modules/ecs_taskdefinition.py)

If these files are inaccurate, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibullbot** [comment]
cc @Java1Guy @jillr @s-hertel @tremble @wimnat
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by alinabuzachis** [comment]
Hi @AlexJov, thank you for raising this issue. It seems this parameter is already supported but it was not properly documented in the Ansible module. dependsOn parameter requires botocore >= 1.3.0. Let me know please if it works for you.  Thank you.
